File: src/main/Java/boj_step4/boj3052.java
import java.io.*;

public class boj3052 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int count = 10;
        boolean[] mod = new boolean[42];
        while (count-- > 0) {
            int dividend = Integer.parseInt(reader.readLine());
            mod[dividend % 42] = true;
        }
        count = 0;
        for (boolean it: mod) if (it) count++;
        writer.append(String.valueOf(count));
        writer.close();
    }
}

File: src/main/kotlin/others/boj3052.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    var modulo42 = IntArray(42)
    var cnt = 10
    var result = 0
    while (cnt-- > 0) {
        val n = r.readLine()!!.toInt()
        modulo42[n % 42]++
    }
    for (i in 0 until modulo42.size)
        if (modulo42[i] != 0) result++
    w.write(result.toString())
    w.close()
}